### PR TITLE
fix(skyrim-platform): optimize texts collection

### LINKED ISSUE #1503: optimize texts collection
https://github.com/skyrim-multiplayer/skymp/blob/82e4d2fc5c4d3601036bd7686f56932260a0cd1a/skyrim-platform/src/platform_se/skyrim_platform/TextsCollection.cpp

Instead of std::unordered_map<int, ...> use std::vector<...>

use [] instead of at (but before using [] check if index is valid)

Use MakeID library (find usage in github search) to allocate/deallocate IDs

### PR DESCRIPTION
**Pull request by ZikkeyLS**
close #1503

### CODE REVIEW

**Comment by ZikkeyLS** [comment]
TO FIX: Currently crushes the game. (I guess because of new exceptions)

**Comment by ZikkeyLS** [comment]
@Pospelove

**Review by Pospelove — CHANGES_REQUESTED** [review]
both set and get methods should throw on failure

File: skyrim-platform/src/platform_se/skyrim_platform/TextsCollection.cpp [hunk 1]
```diff
@@ -16,106 +16,163 @@ int TextsCollection::CreateText(double xPos, double yPos, std::wstring str,
...
-
-  return textCount;
+  texts[id] = text;
+  return id;
 }
 
 void TextsCollection::DestroyText(int textId)
 {
-  texts.erase(textId);
+  if (makeId->IsID(textId) == false) {
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/TextsCollection.cpp:31]: add exception here

**Review by Pospelove — CHANGES_REQUESTED** [review]

**Review by Pospelove — APPROVED** [review]

**Pull request merged**